tuf/tuf.go

```diff
@@ -511,14 +545,15 @@ func (tr *Repo) InitTimestamp() error {
 		return err
 	}
 
-	tr.Timestamp = timestamp
-	return nil
+	return tr.SetTimestamp(timestamp)
 }
 
 // SetRoot sets the Repo.Root field to the SignedRoot object.
 func (tr *Repo) SetRoot(s *data.SignedRoot) error {
 	tr.Root = s
-	return nil
+	var err error
+	tr.originalRootRole, err = tr.Root.BuildBaseRole(data.CanonicalRootRole)
```

Comment:
I think it'd be really helpful for more information on what `originalRootRole` represents for the repo (either here or where it's first declared).  I'm not sure if another name might be more descriptive, I'll let you know if I think of something :)